### ISSUE TITLE
Unicode-containing prefixes on Windows make `api.PrefixData` fail loading prefix state

### ISSUE THREAD

**Issue by jaimergp** [mamba-org/mamba#1201]
`conda` has quite high expectations for Unicode and spaces on paths, and the test suite creates prefixes with all these niceties. While Linux passes the tests alright, Windows [fails](https://github.com/conda/conda/pull/10881/checks?check_run_id=3793321756#step:5:797) with:

```
______________ IntegrationTests.test_clean_tarballs_and_packages ______________
Traceback (most recent call last):
  File "C:\conda_src\tests\test_create.py", line 2461, in test_clean_tarballs_and_packages
    run_command(Commands.REMOVE, prefix, "bzip2", "--yes", "--json")
  File "C:\conda_src\tests\test_create.py", line 293, in run_command
    result = do_call(args, p)
  File "C:\conda_src\conda\cli\conda_argparse.py", line 83, in do_call
    return getattr(module, func_name)(args, parser)
  File "C:\conda_src\conda\cli\main_remove.py", line 90, in execute
    txn = solver.solve_for_transaction()
  File "C:\conda_src\conda\core\solve.py", line 129, in solve_for_transaction
    unlink_precs, link_precs = self.solve_for_diff(update_modifier, deps_modifier,
  File "C:\conda_src\conda\core\solve.py", line 172, in solve_for_diff
    final_precs = self.solve_final_state(update_modifier, deps_modifier, prune, ignore_pinned,
  File "C:\conda_src\conda\core\solve.py", line 1094, in solve_final_state
    state = self._setup_state()
  File "C:\conda_src\conda\core\solve.py", line 1191, in _setup_state
    mamba_prefix_data.load()
RuntimeError: [json.exception.parse_error.101] parse error at line 1, column 1: syntax error while parsing value - unexpected end of input; expected '[', '{', or a literal
```

The last debugging lines are:

```
DEBUG conda.gateways.logging:set_verbosity(231): verbosity set to 3


EXECUTING COMMAND >>> $ conda install --dev -p "D:\Users\jrodriguez\AppData\Local\Temp\1\pytest-of-jrodriguez\pytest-4\test_update_deps_flag_absent0\tmpqznsiqz6\1036 áγђß家固êôōçñ한" flask -y -q


INFO conda.common.io:captured(246): overtaking stderr and stdout
INFO    Loading single package record: "D:\\Users\\jrodriguez\\AppData\\Local\\Temp\\1\\pytest-of-jrodriguez\\pytest-4\\test_update_deps_flag_absent0\\tmpqznsiqz6\\1036 áγђß家固êôōçñ한\\conda-meta\\ca-certificates-2021.7.5-haa95532_1.json"
```

The autogenerated path `"D:\Users\jrodriguez\AppData\Local\Temp\1\pytest-of-jrodriguez\pytest-4\test_update_deps_flag_absent0\tmpqznsiqz6\1036 áγђß家固êôōçñ한"` being the highlight I guess :)

You can see that is not the env creation that fails, but loading state from an already existing prefix to do an additional operation (installing, updating, removing).

**Comment by jaimergp** [comment]
Oh, and the JSON files do exist and have the correct content:

<img width="855" alt="Screenshot 2021-10-05 at 17 55 04" src="https://user-images.githubusercontent.com/2559438/136058841-7ccb6e92-0bfa-4c85-a05b-bc865fbf2048.png">

**Comment by wolfv** [comment]
Cool, thanks for testing this! We'll need to have a look at these

**Comment by wolfv** [comment]
@jaimergp btw looks like conda creates the env with weird chars alright, but then activation doesn't work on Windows :)

I am fixing a couple of Unicode bugs in mamba / micromamba

**Comment by jaimergp** [comment]
Hm, thanks for the report. I don't think this particular test requires activation, so it might have been ignored accidentally. A test in `test_activate.py` might be needed.

**Comment by jaimergp** [comment]
Interestingly I can't create that environment from the CLI either because (I assume) CMD can't handle them well enough and encoding gets lost in the way. But the Python subprocess can. Activation does succeed if only a ñ is contained:

```
(base) D:\Users\jrodriguez>conda activate Aunicodestringñ

(Aunicodestring├▒) D:\Users\jrodriguez>
```

The PS1 is cute though :D

**Comment by wolfv** [comment]
This should be fixed by #1306

**Comment by jaimergp** [comment]
Thanks @wolfv!
